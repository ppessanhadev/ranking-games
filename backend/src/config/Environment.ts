import { injectable } from 'tsyringe';
import { IEnvironment } from '@config/types/IEnvironment';
import 'dotenv/config';

@injectable()
export class Environment implements IEnvironment {
  get get() {
    return {
      port: Number(process.env.PORT) || 4000,
      url: process.env.GAME_URL || 'https://l3-processoseletivo.azurewebsites.net',
    };
  }
}
