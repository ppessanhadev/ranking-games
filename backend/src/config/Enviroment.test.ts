import { Environment } from '@config/Environment';
import 'dotenv/config';

const enviroment = new Environment();

describe('Enviroment', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
  });

  afterAll(() => {
    process.env = OLD_ENV;
  });

  it('should return enviroment values', () => {
    process.env.PORT = '8002';

    expect(enviroment.get.port).toBe(8002);
  });

  it('should return default enviroment values', () => {
    expect(enviroment.get.port).toBe(4000);
  });

  it('should match with snapshot', () => {
    expect(enviroment.get).toMatchSnapshot();
  });
});
