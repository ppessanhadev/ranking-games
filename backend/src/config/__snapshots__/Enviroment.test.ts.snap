// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enviroment should match with snapshot 1`] = `
Object {
  "port": 4000,
  "url": "https://l3-processoseletivo.azurewebsites.net",
}
`;
